src/natcap/invest/sdr/sdr_core.pyx

```diff
@@ -557,4 +567,141 @@ def calculate_sediment_deposition(
+
+        for row_index in range(win_ysize):
+            seed_row = yoff + row_index
+            for col_index in range(win_xsize):
+                flow_sum = 0
+                seed_col = xoff + col_index
+                seed_flow_value = <int>mfd_flow_direction_raster.get(
+                    seed_col, seed_row)
+
+                # Skip this seed if it's nodata.
+                if seed_flow_value == mfd_flow_direction_nodata:
+                    average_flow_raster.set(
+                        seed_col, seed_row, average_flow_nodata)
+                    continue
+
+                # Skip iterating over neighbors if there's no flow at all.
+                if seed_flow_value == 0:
+                    average_flow_raster.set(seed_col, seed_row, 0.0)
+                    continue
+
+                for neighbor_index in range(8):
+                    neighbor_flows[neighbor_index] = 0
+                    neighbor_row = seed_row + ROW_OFFSETS[neighbor_index]
+                    if neighbor_row < 0 or neighbor_row >= n_rows:
+                        continue
+
+                    neighbor_col = seed_col + ROW_OFFSETS[neighbor_index]
+                    if neighbor_col < 0 or neighbor_col >= n_cols:
+                        continue
+
+                    flow_in_direction = (seed_flow_value >> (
+                        neighbor_index * 4) & 0xF)
+                    flow_sum += flow_in_direction
+                    neighbor_flows[neighbor_index] = flow_in_direction
+
+                flow_value_weighted_average = -1.0
+                if flow_sum > 0:
+                    flow_value_weighted_average = 0.0
+                    for neighbor_index in range(8):
+                        neighbor_row = seed_row + ROW_OFFSETS[neighbor_index]
+                        if neighbor_row < 0 or neighbor_row >= n_rows:
+                            continue
+
+                        neighbor_col = seed_col + ROW_OFFSETS[neighbor_index]
+                        if neighbor_col < 0 or neighbor_col >= n_cols:
+                            continue
+
+                        # Flow_sum will always be greater than 0 at this point
+                        # because we check for it up above.
+                        with cython.cdivision(True):
+                            proportional_flow = (
+                                neighbor_flows[neighbor_index] /
+                                (<float>flow_sum))
+
+                        if neighbor_index % 2 == 0:
+                            flow_length = 1.0
+                        else:
+                            flow_length = <float>cmath.M_SQRT2
+
+                        # Flow_length will always be greater than 0 since it
+                        # can only be 1 or sqrt(2).
+                        with cython.cdivision(True):
+                            flow_value_weighted_average += <float>(
+                                (1.0 / flow_length) * proportional_flow)
```

Comment:
Just to save a division here, I'd not calculate `proportional_flow` on line 685 but instead use `neighbor_flows[neighbor_index]` directly in line 699 below and then outside the loop that starts at line 674, divide `flow_value_weighted_average` by `flow_sum`. That'll save you 7 float divisions per cell and be a tich more numerically accurate.

Also changing all the "flow" to "weight" or "aspect" depending on what it is. :)

Reply:
Ah, good call here!  Patched in faeac4ee2d6e25c24793835d9664e5e86dbfd431